src/DebuggerForm.cpp

```diff
@@ -1588,6 +1591,51 @@ void DebuggerForm::processMerge(const QString& message)
 	}
 }
 
+QByteArray DebuggerForm::saveCommands() const
+{
+	QByteArray output;
+
+	auto counter = 0;
+	for (auto &command : commands) {
+		auto [name, description, source, icon, _] = command;
+		if (counter > 0) output.append(':');
+		output.append(name.toUtf8().toBase64());
+		output.append(';');
+		output.append(description.toUtf8().toBase64());
+		output.append(';');
+		output.append(source.toUtf8().toBase64());
+		output.append(';');
+		output.append(icon.toUtf8().toBase64());
+		counter++;
+	}
+
+	return output;
+}
+
+void DebuggerForm::restoreCommands(const QByteArray &input)
+{
+	commands.clear();
+
+	auto commandBytes = input.split(':');
+	auto counter = 0;
+	for (auto& cb: commandBytes) {
+		if (cb.isEmpty()) continue;
+
+		auto fields = cb.split(';');
+		assert(fields.count() == 4);
```

Comment:
Maybe it's better to not assert when `count != 4`, but instead ignore the entry?
This might trigger when a naughty user was editing the saved data ;-)
Or maybe more realistically when a future version of the debugger adds a 5th field, and the user switches between different debugger versions.